### DISCUSSION TITLE
Is p256_dilithium3 supported ?

### DISCUSSION THREAD

**Discussion by maroueneboubakri** [open-quantum-safe/oqs-provider#435] [Q&A]
Hello,

Quick question, is p256_dilithium3 composite supported ? I'm new to this but is there any change to make this combination through the template generator ?

Regards
Maro

**Comment by maroueneboubakri** [comment]
Any support from the maintainers ?

**Comment by baentsch** [comment]
> Any support from the maintainers ?

Make that singular, please :-) I tuned out for a bit...

> is p256_dilithium3 composite supported

Not right away as per https://github.com/open-quantum-safe/oqs-provider?tab=readme-ov-file#signature-algorithms

> is there any change to make this combination through the template generator 

Yes, this should work -- but please note that this would be a hybrid, not a composite (edit/add: Please check the README at the link above to see the differences). If you're _really_ after a Composite, I'd suggest touching base with @feventura who contributed that code).

**Comment by feventura** [comment]
The [Composite draft](https://www.ietf.org/archive/id/draft-ounsworth-pq-composite-sigs-13.html#name-algorithm-identifiers) don't define a dilithium3_p256 combination, the closer you can get is mldsa65_p256.
Saying that, the code for composite can support dilithium instead of mldsa, the only reason I didn't add them is because they are not defined in the composite draft, to add it you will need to insert the combination under [generate.yml](https://github.com/open-quantum-safe/oqs-provider/blob/main/oqs-template/generate.yml#L486) before building the provider.
